pkg/config/operations/permutations/permutations.go

```diff
@@ -104,3 +81,45 @@ func Permute(permutations []Permutation, baseConfig map[string]any) ([]map[strin
 
 	return finalConfigs, err
 }
+
+// applyRelationships iterates over a list of relationships structs and applies them to a config
+func applyRelationships(config map[string]any, relationships []Relationship, keyPathValue any) ([]map[string]any, error) {
```

Comment:
could you leave comments for the logic here and look into the deep copies a little more?

Reply:
Sure. I'll look into the deep copies a bit more.